### PR TITLE
feat: add wallet filter to growers

### LINKED ISSUE #952: Add wallet filter to Growers
Currently there is no way to filter the growers by their wallet 

Wallet is becoming a key way of identifying unique grower actions and for comparing grower activity

### PR DESCRIPTION
**Pull request by henriquemoreiraa**
## Description

Wallet filter in the growers tool

**Issue(s) addressed**

- Resolves #952 

**What kind of change(s) does this PR introduce?**

- [x] Enhancement
- [ ] Bug fix
- [ ] Refactor

**Please check if the PR fulfils these requirements**

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Issue

**What is the current behavior?**

**What is the new behavior?**

## Breaking change

**Does this PR introduce a breaking change?**

## Other useful information
wallet

### CODE REVIEW

File: src/models/FilterGrower.js [hunk 1]
```diff
@@ -16,6 +16,10 @@ export default class Filter {
       where.person_id = this.personId;
     }
 
+    if (this.wallet) {
```
Comment (nmcharlton) [inline src/models/FilterGrower.js:19]: Need to add `wallet` to `countAppliedFilters()` below

File: src/components/FilterTopGrower.js [hunk 2]
```diff
@@ -166,6 +169,16 @@ function FilterTopGrower(props) {
                 onChange={(e) => setEmail(e.target.value)}
                 onKeyDown={handleEnterPress}
               />
+              <TextField
```
Comment (nmcharlton) [inline src/components/FilterTopGrower.js:172]: Can we move this one place up to be after Last Name?
That seems like a more logical location.

**Review by nmcharlton — CHANGES_REQUESTED** [review]

File: src/models/FilterGrower.js (hunk 1 above)
Comment (nmcharlton) [reply src/models/FilterGrower.js:19]: Feel free to refactor `countAppliedFilters()` so we don't have to do this for each new filter 😄 (but no obligation to do so).

File: src/models/FilterGrower.js [hunk 3]
```diff
@@ -57,40 +61,8 @@ export default class Filter {
...
-    }
-
-    if (this.deviceIdentifier) {
-      numFilters += 1;
-    }
-
-    if (this.firstName) {
-      numFilters += 1;
+    if (this.organizationId) {
+      return Object.keys(this.getWhereObj()).length;
```
Comment (nmcharlton) [inline src/models/FilterGrower.js:65]: Note that we need to account for one case where a non-null/undefined (default) value doesn't contribute towards the filter count:
```
this.organizationId === ALL_ORGANIZATIONS
```
See the original implementation:
```
if (this.organizationId && this.organizationId !== ALL_ORGANIZATIONS) {
  numFilters += 1;
}
```

Comment (henriquemoreiraa) [reply src/models/FilterGrower.js:65]: I've made it some tests and when I don't put the if statement, I receive an undefined value for `organization_id`,  so the `Object.keys.length` catches that. In the `getWhereObj` function there isn't a case for `ALL_ORGANIZATIONS`, only if it is different, so when I apply a filter the `organization_id` disappears then it doesn't contribute for the filter count.

[Screencast from 2022-12-12 13-57-29.webm](https://user-images.githubusercontent.com/98126579/207106654-5be4ed86-e0ec-43c4-844e-2bd39636aa49.webm)

Comment (nmcharlton) [reply src/models/FilterGrower.js:65]: Thanks @henriquemoreira, that makes sense. Nice work!

**Review by nmcharlton — APPROVED** [review]

**Pull request merged**